src/detail/flare_actor.cc

```diff
@@ -27,22 +27,27 @@ void flare_actor::act() {
 
 void flare_actor::await_data() {
   BROKER_DEBUG("awaiting data");
+  std::unique_lock<std::mutex> lock{flare_mtx_};
```

Comment:
This would likely work wo/ the lock, but I see you removed the atomic around flare_count and without that, it technically probably isn't guaranteed, so safer to leave in.